draft-ietf-mls-protocol.md

```diff
@@ -1356,7 +1362,41 @@ they should re-use the SignWithLabel construction, using a distinct label.  To
 avoid collisions in these labels, an IANA registry is defined in
 {{mls-signature-labels}}.
 
-The ciphersuites are defined in section {{mls-ciphersuites}}.
+### Public-Key Encryption
+
+As with signing, MLS includes a label and context in encryption operations to
+avoid confusion between ciphertexts produced for different purposes.  Encryption
+and decryption including this label and context are done as follows:
+
+~~~ pseudocode
+EncryptWithLabel(PublicKey, Label, Context, Plaintext) =
+  SealBase(PublicKey, EncryptContext, "", Plaintext)
+
+DecryptWithLabel(PrivateKey, Label, Context, KEMOutput, Ciphertext) =
+  OpenBase(KEMOutput, PrivateKey, EncryptContext, "", Ciphertext)
+~~~
+
+Where EncryptContext is specified as:
+
+~~~ tls
+struct {
+  opaque label<V>;
+  opaque context<V>;
+} EncryptContext;
+~~~
+
+And its fields set to:
+
+~~~
+label = "MLS 1.0 " + Label;
+content = Context;
+~~~
+
+Here, the functions `SealBase` and `OpenBase` are defined {{RFC9180}}, using the
+HPKE algorithms specified by the group's cipehersuite.  If MLS extensions
+require HPKE encryption operations, they should re-use the EncryptWithLabel
+construction, using a distinct label.  To avoid collisions in these labels, an
```

Comment:
Would be possible to recommend or even RECOMMEND that extensions use separate HPKE keys (which can be included in leaves as extensions)?